tools/codegen/dest/lazy_ir.py

```diff
@@ -29,6 +29,9 @@ def node_ctor_inputs(func: LazyIrSchema) -> str:
         else:
             if isinstance(arg.type, VectorCType) and isinstance(arg.type.elem, BaseCType):
                 node_ctor_values.append(f"lazy_tensors::util::ToVector<{arg.type.elem.type}>({arg.name})")
+            elif (isinstance(arg.type, OptionalCType) and
+                  isinstance(arg.type.elem, VectorCType)):
+                node_ctor_values.append(f"lazy_tensors::util::ToOptionalVector<{arg.type.elem.elem.type}>({arg.name})")
```

Comment:
@bdhirsh am I doing this in the least awkward way possible?

Reply:
I think... this is okay haha.

One easy to clean it up a little bit would be to have a function that operates on individual arguments, e.g. from `CType -> str`. Then your funcgtion that maps from `LazyIrSchema` -> "stuff that goes into the constructor" doesn't have to care about what the per-argument mapping looks like.

Another thing that would definitely be more involved, so I'm not sure that it warrants doing now - Ed also gave me a lot of high-level codegen feedback on my recent PR's that probably applies here. `GenLazyNativeFuncDefinition.__call__()` does a lot of stuff and has a lot of string parsing lying around. One example of how to factor the logic some more in this specific case - calling the `Node` constructor - is to write the call as a translation between API's:

```
# this is the kernel that gets registered to the dispatcher
# It has an API corresponding to the dispatcher API
sig = kernel_signature(func, self.backend_index)

# This object knows how to declare/define LazyTensor node constructors
# It follows a different API compared to the dispatcher. For example, it only involves value types.
# And it also requires that some inputs (like vectors and optionals) be transformed by some utils first.
node_ctr_sig = LazyNodeSignature(func) 
{sig.decl(name=f"{self.class_method_name}::{schema.aten_name}")} {{

auto node = node_ctr_sig.inner_call()
}};
```
I ended up with something similar in my functionalization pass codegen [here](https://github.com/pytorch/pytorch/blob/a510f4139b2de1f5baed13bd50a035e39fd2a78b/tools/codegen/api/types.py#L546). Then again, maybe Ed will have some more feedback on it when he gets back :)

Reply:
OK. I at least refactored following the first (simpler) suggestion for now.